sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -180,7 +180,11 @@ def _get_request_data(self, request):
         """
         if request.files:
             form_data = aiohttp.FormData(request.data or {})
-            for form_file, data in request.files.items():
+            if hasattr(request.files, "items"):
+                files = list(request.files.items())
```

Comment:
Why do we need to create a `list`?

Reply:
If each fieldname has only one file, the files type is Dict[str, Tuple]; if multiple files, the files type is List[Tuple] like 
![image](https://github.com/Azure/azure-sdk-for-python/assets/70930885/48f579b6-43c1-474e-8cc1-8a6987c22a4d)
actually the following input format are equal for `requests/https`:
- A: files=[("fieldName", ("filename", file_io, content_type))]
- B: files={"fieldName": ("filename", file_io, content_type)}

We convert dict to list so that the follow-up logic could be unified.